src/lachesis/lachesis.go

```diff
@@ -139,8 +140,8 @@ func (l *Lachesis) initKey() error {
 func (l *Lachesis) initNode() error {
 	key := l.Config.Key
 
-	nodePub := fmt.Sprintf("0x%X", crypto.FromECDSAPub(&key.PublicKey))
-	n, ok := l.Peers.ReadByPubKey(nodePub)
+	nodePub := fmt.Sprintf("0x%X", common.FromECDSAPub(&key.PublicKey))
+	n, ok := l.Peers.ByPubKey[nodePub]
```

Comment:
Why are you reverting @Maxime2's code?

Reply:
Looks like a bug in merging with master, obviously one need to take green line 143 from new code and red line 143 from old code. Otherwise this is place for data race.

Reply:
Yes. Anyway, src/lachesis doesn't work.